### PR TITLE
Catch Up on Notch8 Updates

### PR DESCRIPTION
**Pull request by orangewolf**
- N8 staging set up 
- Overhaul development environment to use more standardized docker process
- Bulkrax install and set up
- Matching Collections during search #224
- Fix Collections metadata
-  Authentication via DB
- Visibility 
- Fix Logout
- Proper seeds file

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -21,10 +21,12 @@
...
+//= require matching_collections_carousel
 
 //= require_directory .
 //= require hyrax
+//= require bulkrax/application
 //= require_tree ./hyrax
 
 //= require browse_everything
-//= require libanswers
+// = require libanswers
```
Comment (orangewolf) [inline app/assets/javascripts/application.js:32]: remove this space please

File: config/initializers/session_store.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Be sure to restart your server when you modify this file.
 
-domain = ["production", "staging","sandbox"].include?(Rails.env) ? '.library.ucsc.edu' : :all
-Rails.application.config.session_store :cookie_store, key: '_ucsc_hyrax_session', domain: domain
+# domain = ["production", "staging","sandbox"].include?(Rails.env) ? '.library.ucsc.edu' : :all
+# Rails.application.config.session_store :cookie_store, key: '_ucsc_hyrax_session', domain: domain
```
Comment (orangewolf) [inline config/initializers/session_store.rb:4]: put this back please

Comment (orangewolf) [reply config/initializers/session_store.rb:4]: replace `.library.ucsc.edu` with `ENV.fetch(SITE_DOMAIN, '.library.ucsc.edu')`

Comment (orangewolf) [reply config/initializers/session_store.rb:4]: and set SITE_DOMAIN for staging

File: config/initializers/sidekiq.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 config = YAML.load(ERB.new(IO.read(Rails.root + 'config' + 'redis.yml')).result)[Rails.env].with_indifferent_access
 
-redis_conn = { url: "redis://#{config[:host]}:#{config[:port]}/" }
+url = Rails.env == 'production' ? "redis://:#{config[:password]}@#{config[:host]}:#{config[:port]}/" : "redis://#{config[:host]}:#{config[:port]}/"
```
Comment (orangewolf) [inline config/initializers/sidekiq.rb:3]: need to resolve this if password is not set

File: stack_car/.sops.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+---
+creation_rules:
+  - pgp: "CHANGEME"
```
Comment (orangewolf) [inline stack_car/.sops.yaml:3]: use or remove

File: stack_car/Gemfile [hunk 5]
```diff
@@ -0,0 +1,154 @@
+source 'https://rubygems.org'
```
Comment (orangewolf) [inline stack_car/Gemfile:1]: remove this file

File: stack_car/ops/staging-deploy.tmpl.yaml [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+    - name: CACHE_SERVER_DERIVATIVE_ENABLED
+      value: "true"
+    - name: CACHE_SERVER_INFO_ENABLED
+      value: "true"
+    - name: CACHE_SERVER_DERIVATIVE
+      value: HeapCache
+    - name: HEAPCACHE_TARGET_SIZE
+      value: 256M
+    - name: CACHE_SERVIER_DERIVATIVE_TTL_SECONDS
+      value: 0
```
Comment (orangewolf) [inline stack_car/ops/staging-deploy.tmpl.yaml:84]: SITE_DOMAIN goes here

Comment (orangewolf) [reply stack_car/ops/staging-deploy.tmpl.yaml:84]: `SITE_DOMAIN=.notch8.cloud`. Mind the `.`

File: app/indexers/collection_indexer.rb [hunk 7]
```diff
@@ -20,6 +20,42 @@ def generate_solr_document
       
       # index the titles a work's ancestor collections
       solr_doc = index_ancestor_titles(solr_doc)
+
+      # index the field that bulkrax uses to keep track of imported/exported records
+      solr_doc[Solrizer.solr_name('bulkrax_identifier', :facetable)] = object.bulkrax_identifier
+
+
+      # index custom fields
```
[the remark this answers is not in the document]
Comment (rmjaffe) [reply app/indexers/collection_indexer.rb:28]: @rschwab Thanks for asking!  Right now, I think the only custom fields that need to be searchable are collectionCallNumber, donorProvenance, subjectName, subjectPlace, subjectTopic, and subjectTitle.  
@JessWaggs: Do recall if we've ever done any real decision making about which metadata properties should be keyword searchable or are the current settings more or less happenstance?  If we haven't, is making those decisions another task that should be included in the Searching & Faceting epic?

File: README.md [hunk 8]
```diff
@@ -1,6 +1,92 @@
...
+- Create a branch
+- Edit
+- Commit
+- `git push n8 <branch-name> ` to push to N8 
+- Merge to default branch (`n8-staging`) to trigger staging deploy
+- When approved in Notch8 QA, make an **MR** against `origin` (using your original, pre n8-staging merge  **MR**)
 
-This project is under heavy development.
+## Log in to a repl on the dev site
+If you need a repl on the dev site, first log in to the webapp container: `docker exec -it hycruz bash`. Then you can just enter `repl` to activate a shortcut I created to set the bundle parameters correctly and start the repl.
```
[the remark this answers is not in the document]
Comment (bkiahstroud) [reply README.md:92]: The N8 stuff definitely needs to come out. For the setup instructions: 

1. Should we have two setup sections (one with stack_car and one without)? Or 
2. Should the stack_car setup instructions be in a separate MD file?

[the remark this answers is not in the document]
Comment (rschwab) [reply README.md:92]: I think having two setup sections sounds good. @snehagunduraoUL what do you think?

File: app/views/hyrax/collections/_show_collection_without_subcollections.html.erb [hunk 9]
```diff
@@ -32,6 +32,10 @@
     </div>
     <div class="collection-text">
       <%= render 'collection_description', presenter: @presenter %>
+      <% if @presenter.harmfulLanguageStatement.present?%>
+        <h5 class='harmful-language-statement'><b><%=t('hyrax.collections.show.harmful_language_statement_heading')%></b></h5>
```
[the remark this answers is not in the document]
Comment (rschwab) [reply app/views/hyrax/collections/_show_collection_without_subcollections.html.erb:36]: @bkiahstroud didn't have write access to your remote so a patch may be the easier route at this point
[0001-Remove-bold-tag-in-favor-of-css.patch.txt](https://github.com/UCSCLibrary/ucsc-library-digital-collections/files/7507153/0001-Remove-bold-tag-in-favor-of-css.patch.txt)
.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (snehagunduraoUL) [reply README.md:92]: Agree!

**Comment by snehagunduraoUL** [comment]
probably a good idea to remove all helm/kubernetes and .gitlab files?

File: .gitlab-ci.yml [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+  when: manual
+  allow_failure: true
+  script:
+    - export KUBECONFIG=$KUBECONFIG_UCSC
+    - ./bin/helm_delete ucsc-hyrax-staging ucsc-hyrax-staging
+  tags:
+    - local
+
+.deploy:
+  image: dtzar/helm-kubectl:3.5.3
```
[the remark this answers is not in the document]
Comment (bkiahstroud) [reply .gitlab-ci.yml:116]: This file (and the helm/kubernetes stuff) are what let us deploy to our staging. If we remove it we will need access to your staging. 

The other .gitlab stuff can be removed without issue

[the remark this answers is not in the document]
Comment (snehagunduraoUL) [reply .gitlab-ci.yml:116]: @bkiahstroud Sure. we can retain the helm files. Let's remove just the gitlab stuff for now. Thank you.

File: app/indexers/collection_indexer.rb (hunk 7 above)
[the remark this answers is not in the document]
Comment (rschwab) [reply app/indexers/collection_indexer.rb:28]: @rmjaffe I've added [480 - Determine which custom fields will be keyword searchable](https://github.com/UCSCLibrary/dams_project_mgmt/issues/480) to handle this.

**Pull request merged**